addons/pos_restaurant/static/src/js/Screens/SplitBillScreen/SplitBillScreen.js

```diff
@@ -96,29 +96,11 @@ odoo.define('pos_restaurant.SplitBillScreen', function(require) {
         _splitQuantity(line) {
             const split = this.splitlines[line.id];
 
-            let totalQuantity = 0;
-
-            this.env.pos.get_order().get_orderlines().forEach(function(orderLine) {
-                if(orderLine.get_product().id === split.product)
-                    totalQuantity += orderLine.get_quantity();
-            });
-
             if(line.get_quantity() > 0) {
-                if (!line.get_unit().is_pos_groupable) {
-                    if (split.quantity !== line.get_quantity()) {
-                        split.quantity = line.get_quantity();
-                    } else {
-                        split.quantity = 0;
-                    }
+                if (split.quantity < line.get_quantity()) {
+                    split.quantity += line.get_unit().is_pos_groupable? 1: line.get_unit().rounding;
```

Comment:
```suggestion
                const lineQty = line.get_quantity();
                if (split.quantity < lineQty) {
                    split.quantity += line.get_unit().is_pos_groupable? 1: line.get_unit().rounding;
                    if (split.quantity > lineQty) {
                        split.quantity = lineQty;
                    }
```
Imagine you have a product in unit and you have 1.5 of this product, if you split and click once it will be marked as 1/1.5 and if you click two times it will be put 2/1.5 and we do not want to go further than the maximum amout

Reply:
Also, why do you remove the thing where, with non pos groupable units, you cannot split it ? I think it is the principle behind non-groupable units, the product is indivisible and ungroupable with other.